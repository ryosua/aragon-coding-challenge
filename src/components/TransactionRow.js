import React from 'react'
import PropTypes from 'prop-types'
import { TableRow, TableCell, Text } from '@aragon/ui'

const TransactionRow = ({ hash, from, to, value }) => (
    <TableRow>
        <TableCell>
            <Text>{hash}</Text>
        </TableCell>
        <TableCell>
            <Text>{from.address}</Text>
        </TableCell>
        <TableCell>
            <Text>{to.address}</Text>
        </TableCell>
        <TableCell>
            <Text>{value}</Text>
        </TableCell>
    </TableRow>
)

TransactionRow.propTypes = {
    hash: PropTypes.string.isRequired,
    from: PropTypes.object.isRequired,
    to: PropTypes.object.isRequired,
    value: PropTypes.number.isRequired
}

export default TransactionRow
