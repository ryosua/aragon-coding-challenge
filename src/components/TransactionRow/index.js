import React from 'react'
import PropTypes from 'prop-types'
import { TableRow, TableCell, Text } from '@aragon/ui'
import { convertWeiToUSD } from 'api/getConversionRate'
import RateContext from 'contexts/RateContext'

const TransactionRow = ({ hash, from, to, value }) => {
    return (
        <RateContext.Consumer>
            {({ ethToUSDRate }) => {
                const valueInUSD = ethToUSDRate ? convertWeiToUSD(value, ethToUSDRate) : 'loading'
                return (
                    <TableRow>
                        <TableCell>
                            <Text>{hash}</Text>
                        </TableCell>
                        <TableCell>
                            <Text>{from.address}</Text>
                        </TableCell>
                        <TableCell>
                            <Text>{to.address}</Text>
                        </TableCell>
                        <TableCell>
                            <Text>{valueInUSD}</Text>
                        </TableCell>
                    </TableRow>
                )
            }}
        </RateContext.Consumer>
    )
}

TransactionRow.propTypes = {
    hash: PropTypes.string.isRequired,
    from: PropTypes.object.isRequired,
    to: PropTypes.object.isRequired,
    value: PropTypes.number.isRequired
}

export default TransactionRow
