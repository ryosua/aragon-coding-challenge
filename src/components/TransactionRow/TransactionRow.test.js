import React from 'react'
import renderer from 'react-test-renderer'
import TransactionRow from './index.js'

it('TransactionRow renders according to snapshot.', () => {
    const transaction = {
        hash: '0xd67348d36f906ba7b2be09baa868e0e93cf310f8d06146ec8e0936e96410e05d',

        from: {
            address: '0xBA826fEc90CEFdf6706858E5FbaFcb27A290Fbe0'
        },
        to: {
            address: '0xD0a4b8946Cb52f0661273bfbC6fD0E0C75Fc6433'
        },
        value: 0
    }

    const tree = renderer.create(<TransactionRow {...transaction} />).toJSON()
    expect(tree).toMatchSnapshot()
})
