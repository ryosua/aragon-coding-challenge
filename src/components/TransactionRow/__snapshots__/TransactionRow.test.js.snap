// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionRow renders according to snapshot. 1`] = `
<tr
  className="TableRow__StyledTableRow-sc-3jn6yz-0 cFtmyC"
>
  <td
    className="TableCell__StyledTableCell-sc-110j155-0 bqABXJ"
  >
    <div
      className="TableCell__StyledTableCellContent-sc-110j155-1 klobR"
    >
      <span
        className="Text-yxw4o9-0 bIIgvx"
      >
        0xd67348d36f906ba7b2be09baa868e0e93cf310f8d06146ec8e0936e96410e05d
      </span>
    </div>
  </td>
  <td
    className="TableCell__StyledTableCell-sc-110j155-0 bqABXJ"
  >
    <div
      className="TableCell__StyledTableCellContent-sc-110j155-1 klobR"
    >
      <span
        className="Text-yxw4o9-0 bIIgvx"
      >
        0xBA826fEc90CEFdf6706858E5FbaFcb27A290Fbe0
      </span>
    </div>
  </td>
  <td
    className="TableCell__StyledTableCell-sc-110j155-0 bqABXJ"
  >
    <div
      className="TableCell__StyledTableCellContent-sc-110j155-1 klobR"
    >
      <span
        className="Text-yxw4o9-0 bIIgvx"
      >
        0xD0a4b8946Cb52f0661273bfbC6fD0E0C75Fc6433
      </span>
    </div>
  </td>
  <td
    className="TableCell__StyledTableCell-sc-110j155-0 bqABXJ"
  >
    <div
      className="TableCell__StyledTableCellContent-sc-110j155-1 klobR"
    >
      <span
        className="Text-yxw4o9-0 bIIgvx"
      >
        loading
      </span>
    </div>
  </td>
</tr>
`;
